src/components/ListSection/ListSection.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+
+function ListSection({ variant, as, children }) {
+    return (
+        <StyledListSection $variant={variant} $as={as}>
+            {children}
+        </StyledListSection>
+    );
+}
+
+ListSection.propTypes = {
+    variant: PropTypes.string,
+    as: PropTypes.string,
+    children: PropTypes.node,
+};
+
+const StyledListSection = styled.div`
+    background-color: ${({ $variant }) => ($variant === 'dark' ? 'var(--dark-color-2)' : 'var(--color-2)')};
+    padding: 1em;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    flex: 1;
+    // overflow hidden nie działało
```

Comment:
Niestety nie mogę tego odtworzyć problemu, ale może to będzie rozwiązaniem:
https://stackoverflow.com/questions/39420256/border-radius-disappears-with-overflow-rule